### ISSUE TITLE
RFE: Support for App Widget (Android)

### ISSUE THREAD

**Issue by kutoman** [codenameone/CodenameOne#2885]
This is the Android equivalent of #2777. As it was for the iOS version, this description is also not exhaustive since it will be based on my integration which is limited to the requirements I had in my project. 
The Android widget is defined by its meta file (XML file placed in the "xml" subfolder of "res") and its layout file (XML file placed in the "layout" subfolder of "res"). The mentioned meta file contains a reference to a resource image file which is later used as a thumbnail in Android's widget menu. The widget's mechanics / implementation is just done in classes (extending `AppWidgetProvider`) as we're used to for the implementation of the rest of the app.  Of course, to finally add the widget to the app, the widget's meta file needs to be referenced in the manifest as a `<receiver>`. 

Currently it is possible to integrate the widget classes in CN1 but not the widgets resource files.

**Comment by codenameone** [comment]
I think that in android you can package the widget as a library in an AAR and integrate that into the project.

**Comment by kutoman** [comment]
@codenameone that approach seems to be cumbersome if the aar library needs to be also dependent on the main project. In my case, for example, I use some classes which are part of the main project and thus shared implementations. On my attempt to move the widget in a separate library module (to build the aar) I get the issue of a **Circular Dependency**

**Comment by codenameone** [comment]
Yes it's a problem but unfortunately there's no easy solution as widgets aren't portable. You would need to write the native widget with no circular dependencies. Yes it will create some duplication of effort but because widgets are relatively simple hopefully there shouldn't be too much.

The only alternative I can think of is that you might be able to define widgets by injecting into our manifest the widget details and include the widget native code directly in the native directory. This might be cumbersome to build/debug but might be more appropriate for your needs.

**Comment by kutoman** [comment]
The **aar** library approach seems to be legit with respect to Android app widget integration. Instead of having all widget specific files in the **aar** I decided to just put the resource files in there and all the rest is kept in the CN1 project directly (actual code, references in the manifest etc). This way I don't need to worry about having duplication or circular dependencies in the **aar** library. This works quite well. 
I guess this issue can be closed now as CN1 is kinda supporting widget integration this way in terms of android at least.
